### DISCUSSION TITLE
Rectangle Pro To-Do Mode?

### DISCUSSION THREAD

**Discussion by hankleukart** [rxhanson/RectanglePro-Community#4] [Q&A] [answered]
Hey Ryan! Is there a way to duplicate the Rectangle To-Do Mode in Rectangle Pro? I have an ultrawide monitor, and I use the original Rectangle to keep iMessage as a small strip on the right side, and then the remaining screen real estate is divided into thirds. I can't seem to do this with Rectangle Pro:

![to-do-mode](https://user-images.githubusercontent.com/97927016/149857432-280a7fe9-32a8-4d8f-b93d-a6542eb0b584.png)

I also wish there were an easy way to tile two small windows inside the right-side To-Do Mode strip (so I could put Messages and a Task List over there) -- so the old To-Do Mode isn't perfect, but it least it enables the Messages scenario.

![to-do-mode-tiled](https://user-images.githubusercontent.com/97927016/149857439-99f1eea1-7380-4080-8035-7be7ea50a511.png)

-Hank

**Comment by rxhanson** [comment, marked as answer]
Thanks for using Rectangle Pro! I just released v2.1 a couple minutes ago that has the Todo mode stuff in the settings. It was previously only enabled via command line. Note that I renamed Todo mode to Pin mode in Rectangle Pro.

I have actually thought about this a lot, and was originally going to hold out on adding Todo mode into Pro until I built something that was more akin to an actual "Tiling" window manager, but I decided to just go ahead and release what is there with the rename to Pin mode, which is a little more flexible naming for future updates if I plan to make changes to accommodate what you're looking for. 

I'm still not set on what I want to do here. I will consider just making it what you're looking for, or I might stretch it and go all the way for more serious tiling capabilities.

**Comment by hankleukart** [comment]
Wow, thanks! I'll grab it and definitely pay the $10 because this might be the most useful MacOS-improving app I have (and for anyone that has the Samsung Odyssey G9 Ultrawide).

Yeah, I agree that the way Todo Mode works today doesn't feel like the cleanest design in the long run even though I'm glad I have it as a stopgap. It's hard to decide what the right design is, because one of the biggest assets of Rectangle is that it's super simple, so its learning curve is tiny and it's very easy to use. I suppose one design would be to let users designate any number of windows (docked against any sides or longwise-sides of the screen?) as "permanent" or "out of bounds," so all other half/third/fourth actions only use the remaining available space. But I can also imagining implementing something like i3, though that definitely increases the complexity for most average MacOS users.
-Hank
